spec/enumerable_spec.rb

```diff
@@ -0,0 +1,244 @@
+require_relative '../enumerable'
+
+describe 'enumerable' do
+  describe '#my_each' do
+    let(:item) { [1, 2, 3, 4] }
+
+    it 'returns an enumerator with the expected values when called without arguments' do
+      expect(item.my_each.class).to eql(item.each.class)
+      expect(item.my_each.to_a).to match_array(item.each.to_a)
+    end
+
+    it 'returns the expected result when passed a block' do
+      expected = []
+      actual = []
+      item.each { |element| expected << element + 10 }
+      item.my_each { |element| actual << element + 10 }
+      expect(expected).to match_array(actual)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    let(:item) { [1, 2, 3, 4] }
+    it 'returns an enumerator with the expected values when called without arguments' do
+      expect(item.my_each_with_index.class).to eql(item.each_with_index.class)
+      expect(item.my_each_with_index.to_a).to match_array(item.each_with_index.to_a)
+    end
+
+    it 'returns the expected result when passed a block' do
+      expected = {}
+      actual = {}
+      item.each_with_index { |item, index| expected[index] = item + 10 }
+      item.my_each_with_index { |item, index| actual[index] = item + 10 }
+      expect(expected).to match(actual)
+    end
+  end
+
+  describe '#my_select' do
+    let(:item) { [1, 2, 3, 4] }
+    it 'returns an enumerator with the expected values when called without arguments' do
+      expect(item.my_select.class).to eql(item.select.class)
+      expect(item.my_select.to_a).to match_array(item.select.to_a)
+    end
+
+    it 'returns the expected result when passed a block' do
+      expect(item.select(&:even?)).to match_array(item.my_select(&:even?))
+    end
+  end
+
+  describe '#my_all' do
+    let(:item) { %w[ant bear cat] }
```

Comment:
- Kindly move this ```let(:item) { %w[ant bear cat] }``` line and place it between lines 3 and 4 thus making it a global variable. Then, remove it from all other places they currently exist.